smart-hadoop-support/smart-hadoop/src/test/java/org/smartdata/hdfs/action/TestCopy2S3Action.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata.hdfs.action;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.s3a.S3AFileSystem;
+import org.apache.hadoop.hdfs.DFSTestUtil;
+import org.junit.Assert;
+import org.junit.Test;
+import org.smartdata.SmartContext;
+import org.smartdata.action.MockActionStatusReporter;
+import org.smartdata.conf.SmartConf;
+import org.smartdata.hdfs.HadoopUtil;
+import org.smartdata.hdfs.MiniClusterHarness;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Test for Copy2S3FileAction.
+ */
+public class TestCopy2S3Action extends MiniClusterHarness {
+  //change this to connect with aws s3
+  final String S3_ACCESS_KEY = "AKIAJ35KYRLMT2CFVXWQ";
+  final String S3_SECRET_KEY = "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";
+
+  private SmartContext smartContext1;
+
+  private void copy2S3File(String src, String dest, long offset) throws Exception {
+    Copy2S3Action copy2S3Action = new Copy2S3Action();
+    copy2S3Action.setDfsClient(dfsClient);
+    SmartConf configuration = smartContext.getConf();
+    smartContext1 = smartContext;
+    configuration.set("fs.s3a.access.key", S3_ACCESS_KEY);
+    configuration.set("fs.s3a.secret.key", S3_SECRET_KEY);
+    smartContext1.setConf(configuration);
+    copy2S3Action.setContext(smartContext1);
+    copy2S3Action.setStatusReporter(new MockActionStatusReporter());
+    Map<String, String> args = new HashMap<>();
+    args.put(CopyFileAction.FILE_PATH, src);
+    args.put(CopyFileAction.DEST_PATH, dest);
+    args.put(CopyFileAction.OFFSET_INDEX, "" + offset);
+    copy2S3Action.init(args);
+    copy2S3Action.run();
+  }
+
+  @Test
+  public void testS3FileCopy() throws Exception {
+    final String srcPath = "/testCopy";
+    final String file1 = "file1";
```

Comment:
We have only one file in this UT. Please choose a meaningful name for this parameter.